pkg/assembler/nodes.go

```diff
@@ -16,9 +16,161 @@
 package assembler
 
 // ArtifactNode is a node that represents an artifact
-// TODO: implement GuacNode
-type ArtifactNode struct{}
+type ArtifactNode struct {
+	Name     string
+	Checksum string
+}
+
+func (an ArtifactNode) Type() string {
+	return "Artifact"
+}
+
+func (an ArtifactNode) Properties() map[string]interface{} {
+	properties := make(map[string]interface{})
+	properties["name"] = an.Name
+	properties["checksum"] = an.Checksum
+	return properties
+}
+
+func (an ArtifactNode) Attributes() []string {
+	return []string{"name", "checksum"}
```

Comment:
I wonder if there is a way to look this up automatically? It seems like a future maintenance burden to need to know that you need to set `"name"` and `"checksum"` in two different places.

Reply:
Agree, this should be possible the same the json serialization/deserialization is done via tags on struct fields, but I didn't get a chance to play with those yet, punting for another PR.

Added a TODO